### ISSUE TITLE
PySpark error - "OSError: [Errno 2] No such file or directory"

### ISSUE THREAD

**Issue by dmcclory** [spotify/luigi#1070]
I'm trying to run a PySparkTask from luigi, but I've been hitting this stack trace. I'm just trying to run it locally for now:

```
INFO: [pid 45554] Worker Worker(salt=641770409, workers=1, host=dans-air, username=dan, pid=45554) running   PlanDesignBuilder(design_output=s3://picwell.vendors.promptworks/luigi_spike/trial_benefit_output.json)
INFO: Running: ['spark-submit', '--deploy-mode', 'client', '--name', 'PlanDesignBuilder', '/Users/dan/prompt/picwell/etl/env/lib/python2.7/site-packages/luigi/contrib/pyspark_runner.py', '/var/folders/j5/ng5x9x4x4txgf82ykr7c_wlc0000gn/T/PlanDesignBuilderLZSP4w/PlanDesignBuilder.pickle']
ERROR: [pid 45554] Worker Worker(salt=641770409, workers=1, host=dans-air, username=dan, pid=45554) failed    PlanDesignBuilder(design_output=s3://picwell.vendors.promptworks/luigi_spike/trial_benefit_output.json)
Traceback (most recent call last):
  File "/Users/dan/prompt/picwell/etl/env/lib/python2.7/site-packages/luigi/worker.py", line 137, in run
    new_deps = self._run_get_new_deps()
  File "/Users/dan/prompt/picwell/etl/env/lib/python2.7/site-packages/luigi/worker.py", line 88, in _run_get_new_deps
    task_gen = self.task.run()
  File "/Users/dan/prompt/picwell/etl/env/lib/python2.7/site-packages/luigi/contrib/spark.py", line 336, in run
    super(PySparkTask, self).run()
  File "/Users/dan/prompt/picwell/etl/env/lib/python2.7/site-packages/luigi/contrib/spark.py", line 235, in run
    universal_newlines=True)
  File "/Users/dan/.pyenv/versions/2.7.5/lib/python2.7/subprocess.py", line 711, in __init__
    errread, errwrite)
  File "/Users/dan/.pyenv/versions/2.7.5/lib/python2.7/subprocess.py", line 1308, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
```

luigi.conf:
```
[spark]
spark-submit: /Users/dan/prompt/picwell/spark-1.4.0-bin-hadoop2.6/bin/spark-submit
master: spark://localhost:7077
```

This has been happening with luigi at 1.3.0, on spark 1.3 and 1.4.

Issue #866 had a similar stack trace, but the proposed solution (making sure that the name parameter doesn't have spaces) doesn't apply here.

I'm happy to provide any other information! Does this seem like a bug or user error?

**Comment by jthi3rry** [comment]
Hi @dmcclory,

Your config seems good, but the task isn't picking up the values for spark-submit and master.

Is your config file actually called `luigi.conf`? It would explain why luigi isn't reading it. Try renaming it to `luigi.cfg` or `client.cfg` as per http://luigi.readthedocs.org/en/latest/configuration.html.

With your config, the line
```INFO: Running: ['spark-submit', '--deploy-mode', 'client',  ... ]```

Should read:
```INFO: Running: ['/Users/dan/prompt/picwell/spark-1.4.0-bin-hadoop2.6/bin/spark-submit', '--master', 'spark://localhost:7077', '--deploy-mode', 'client', ... ]```

**Comment by dmcclory** [comment]
Sorry, I wrote `luigi.conf`, but my config file was actually named `luigi.cfg`. 

I looked at the run method under a debugger, comparing the `luigi.cfg` and `client.cfg` cases. When the config is named `client.cfg`, `self.spark_command` has the right value.

But when the config is named `luigi.cfg`, `self.spark_command` returns `'spark-submit'`, which is the default value in a configuration lookup. The call to `configuration.get_config()` in `spark_submit` is returning an empty LuigiConfigParser.
